.travis.yml

```diff
@@ -1,23 +1,38 @@
+dist: trusty
+sudo: false
+
 language: python
 
 cache:
-  apt: true
-  # We use three different cache directory
-  # to work around a Travis bug with multi-platform cache
   directories:
   - $HOME/.cache/pip
-  - $HOME/download
-env:
-  global:
-    # Directory where tests are run from
-    - TEST_DIR=/tmp/test_dir/
-    - MODULE=skltemplate
-  matrix:
-    - DISTRIB="conda" PYTHON_VERSION="2.7"
-      NUMPY_VERSION="1.8.2" SCIPY_VERSION="0.13.03" CYTHON_VERSION="0.21"
-    - DISTRIB="conda" PYTHON_VERSION="3.5" COVERAGE="true"
-      NUMPY_VERSION="1.14.0" SCIPY_VERSION="1.0.0" CYTHON_VERSION="0.27.3"
 
-install: source ci_scripts/travis/install.sh
-script: bash ci_scripts/travis/test.sh
-after_success: source ci_scripts/travis/success.sh
+matrix:
+  include:
+    - env: PYTHON_VERSION="2.7"
+    - env: PYTHON_VERSION="3.5"
+    - env: PYTHON_VERSION="3.6"
+    - env: PYTHON_VERSION="3.7"
+
+install:
+  # install miniconda
+  - deactivate
+  - wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
+  - MINICONDA_PATH=/home/travis/miniconda
+  - chmod +x miniconda.sh && ./miniconda.sh -b -p $MINICONDA_PATH
+  - export PATH=$MINICONDA_PATH/bin:$PATH
+  - conda update --yes conda
+  # create the testing environment
+  - conda create -n testenv --yes python=$PYTHON_VERSION pip
+  - source activate testenv
+  - conda install --yes numpy scipy pytest pytest-cov
+  - pip install --pre scikit-learn
+  - pip install codecov
+  # install the contribution
+  - pip install .
+
+script:
+  - pytest -lv --cov-report term-missing skltemplate --cov=skltemplate
```

Comment:
Maybe just use the default settings + coverage, and let people customize if they want to,
```
pytest -v --cov=skltemplate --pyargs skltemplate
```